app/uk/gov/hmrc/customs/declaration/connectors/upscan/UpscanInitiateConnector.scala

```diff
@@ -34,20 +33,21 @@ class UpscanInitiateConnector @Inject()(http: HttpClient,
                                         config: DeclarationsConfigService)
                                        (implicit ec: ExecutionContext) {
 
-  def send[A](payload: UpscanInitiatePayload, apiVersion: ApiVersion)(implicit vfupr: ValidatedFileUploadPayloadRequest[A]): Future[UpscanInitiateResponsePayload] = {
+  def send[A](payload: UpscanInitiatePayload, apiVersion: ApiVersion)(implicit vfupr: ValidatedFileUploadPayloadRequest[A], hc: HeaderCarrier): Future[UpscanInitiateResponsePayload] = {
     if (payload.isV2) {
       post(payload, config.fileUploadConfig.upscanInitiateV2Url)
     } else {
       post(payload, config.fileUploadConfig.upscanInitiateV1Url)
     }
   }
 
-  private def post[A](payload: UpscanInitiatePayload, url: String)(implicit vfupr: ValidatedFileUploadPayloadRequest[A]) = {
+  private def post[A](payload: UpscanInitiatePayload, url: String)(implicit vfupr: ValidatedFileUploadPayloadRequest[A], hc: HeaderCarrier) = {
 
-    implicit val hc: HeaderCarrier = HeaderCarrier()
+    implicit val headerCarrier: HeaderCarrier = HeaderCarrier()
+    val upscanHeader: Seq[(String, String)] = hc.headers(List("Accept", "Gov-Test-Scenario"))
```

Comment:
Here is the third usage of this method to retrieve the stub headers. Please unify them somehow.